operators/src/plot/statistics.rs

```diff
@@ -0,0 +1,261 @@
+    Operator, PlotOperator, PlotQueryProcessor, PlotResultDescriptor, QueryContext, QueryProcessor,
+    QueryRectangle, TypedPlotQueryProcessor, TypedRasterQueryProcessor,
+};
+use crate::error;
+use crate::util::number_statistics::NumberStatistics;
+use crate::util::Result;
+
+/// A plot that outputs basic statistics about its inputs
+///
+/// Does currently not use a weighted computations, so it assumes equally weighted
+/// time steps in the sources.
+pub type Statistics = Operator<StatisticsParams>;
+
+/// The parameter spec for `Statistics`
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct StatisticsParams {}
+
+#[typetag::serde]
+impl PlotOperator for Statistics {
+    fn initialize(
+        self: Box<Self>,
+        context: &dyn ExecutionContext,
+    ) -> Result<Box<InitializedPlotOperator>> {
+        // TODO: implement operator also for vector data
+        ensure!(
+            self.vector_sources.is_empty(),
+            error::InvalidNumberOfVectorInputs {
+                expected: 0..1,
+                found: self.vector_sources.len()
+            }
+        );
+
+        Ok(InitializedStatistics {
+            params: (),
+            result_descriptor: PlotResultDescriptor {},
+            raster_sources: self
+                .raster_sources
+                .into_iter()
+                .map(|o| o.initialize(context))
+                .collect::<Result<Vec<_>>>()?,
+            vector_sources: self
+                .vector_sources
+                .into_iter()
+                .map(|o| o.initialize(context))
+                .collect::<Result<Vec<_>>>()?,
+            state: (),
+        }
+        .boxed())
+    }
+}
+
+/// The initialization of `Statistics`
+pub type InitializedStatistics = InitializedOperatorImpl<(), PlotResultDescriptor, ()>;
+
+impl InitializedOperator<PlotResultDescriptor, TypedPlotQueryProcessor> for InitializedStatistics {
+    fn query_processor(&self) -> Result<TypedPlotQueryProcessor> {
+        Ok(TypedPlotQueryProcessor::JSON(
+            StatisticsQueryProcessor {
+                rasters: self
+                    .raster_sources
+                    .iter()
+                    .map(|source| source.query_processor())
+                    .collect::<Result<Vec<_>>>()?,
+            }
+            .boxed(),
+        ))
+    }
+}
+
+/// A query processor that calculates the statistics about its inputs.
+pub struct StatisticsQueryProcessor {
+    rasters: Vec<TypedRasterQueryProcessor>,
+}
+
+impl PlotQueryProcessor for StatisticsQueryProcessor {
+    type PlotType = serde_json::Value;
+
+    fn plot_query<'a>(
+        &'a self,
+        query: QueryRectangle,
+        ctx: &'a dyn QueryContext,
+    ) -> BoxFuture<'a, Result<Self::PlotType>> {
+        let mut queries = Vec::with_capacity(self.rasters.len());
+        for (i, raster_processor) in self.rasters.iter().enumerate() {
```

Comment:
maybe it would make sense to process the childs in parallel. Always continuing with the source that has the next tile available the earliest using the select macro

Reply:
I agree, but we first have to find a way to incorporate our pool in queries :smile: